gui/src/pluginmanager.cpp

```diff
@@ -3947,7 +3947,7 @@ void CatalogMgrPanel::OnTarballButton(wxCommandEvent& event) {
   PluginMetadata metadata;
   bool ok = handler->ExtractMetadata(path.ToStdString(), metadata);
   if (!ok) {
-    OCPNMessageBox(this, _("Error extracting metadata from tarball."),
+    OCPNMessageBox(this, _("Cannot load plugin, tarball does not contain metadata.xml file."),
```

Comment:
This is based on your personal experience, you don't know it's missing at this point.

Perhaps: "Error extracting metadata from tarball (missing metadata?)"

Reply:
True, the tarball could be malformed, or not enough disk space, or the permissions are invalid.

Reply:
How about changing `ExtractMetadata` to return an integer value? This would make it possible to print a detailed error message.

```c++
// Define error codes
enum MetadataExtractionError {
  NO_ERROR = 0,
  FILE_NOT_FOUND = 1,
  INVALID_TARBALL = 2,
  MISSING_METADATA_XML = 3,
  PARSE_ERROR = 4,
  // Add more error codes as needed
};

// Function to get localized error message
wxString GetLocalizedErrorMessage(MetadataExtractionError error) {
  switch (error) {
    case FILE_NOT_FOUND:
      return _("File not found");
    case INVALID_TARBALL:
      return _("Invalid tarball format");
    case MISSING_METADATA_XML:
      return _("Missing metadata.xml in tarball");
    case PARSE_ERROR:
      return _("Error parsing metadata");
    default:
      return _("Unknown error");
  }
}
```

Reply:
Perhaps more work and code then this is worth. I think we should just modify the test string for now and be done with this for now.

On a sidenote, plain `enum` is also a bad thing since the constants are global. ` enum class` is better, this has a local name scope for the constants. Look for `enum class` to find examples.

Reply:
ok sure. I will just change the message as you have suggested.